table/internal_iterator.h

```diff
@@ -17,6 +17,11 @@ namespace rocksdb {
 
 class PinnedIteratorsManager;
 
+struct IterateResult {
+  Slice key = Slice();
+  bool may_be_out_of_upper_bound = true;
```

Comment:
I'm paranoid but do we have to initial the values, since we are going to set them anyway?